lib/engine/game/g_1858/game.rb

```diff
@@ -397,6 +511,48 @@ def payout_companies
           super(ignore: exchanged_companies.map(&:id))
         end
 
+        # Removes all of the icons on the map for a private railway company.
+        def delete_icons(company)
+          icon_name = company.sym.delete('&')
+          @hexes.each do |hex|
+            hex.tile.icons.delete_if { |icon| icon.name == icon_name }
```

Comment:
always use regex

Reply:
@tobymao  I don't understand what you mean here. Use a regexp for testing the strings are the same? Something like:

```
rx = /^#{company.sym.delete('&')}$/
@hexes.each do |hex|
  hex.tile.icons.delete_if { |icon| rx.match?(icon.name) }
end
```
Is there an issue using `==` to test for string equality? AFAIK that works in both Ruby and Javascript.

Reply:
reject… sorry typo

Reply:
Ah, that makes _much_ more sense. I've changed the `Array.delete_if` call to `Array.reject!`.